tidy3d/web/core/task_core.py

```diff
@@ -472,18 +472,36 @@ def get_sim_data_hdf5(
         if not self.task_id:
             raise WebError("Expected field 'task_id' is unset.")
 
+        file = None
         try:
-            return download_file(
-                self.task_id,
-                SIMULATION_DATA_HDF5,
+            file = download_gz_file(
+                resource_id=self.task_id,
+                remote_filename=SIMULATION_DATA_HDF5_GZ,
                 to_file=to_file,
                 verbose=verbose,
                 progress_callback=progress_callback,
             )
-        except Exception:
-            raise WebError(
-                f"Failed to download file '{SIMULATION_DATA_HDF5}' from server. Please confirm that the task was successful."
-            )
+        except ClientError:
+            if verbose:
+                console = get_logger_console()
+                console.log(f"Unable to download '{SIMULATION_DATA_HDF5_GZ}'.")
+
+        if not file:
+            try:
```

Comment:
I think the difference from other places `download_file` is used is that here the error message asks to check if the task was successful. Maybe I should change this to `successfully run`?

Reply:
yea I think that would make sense

Reply:
Ok, I did that. The reason this part (and even the lines around it) of the code is replicated in `web/core/task_core.py` and in `web/api/mode.py` is simply because we need to unify the mode solver to be handled like FDTD and HEAT currently are. There's actually a lot of overlapping code between these two files...

By the way, when doing that, we should also still implement solver-specific handling. Currently web api for heat is unified with fdtd without any separate e.g. `taskInfo` classes, and without too much branching, but one reason this is possible is that e.g. on the backend we have to make sure that we use the same solver data file name for both solvers (e.g. `simulation_data.hdf5.gz` in this PR). Generally different solvers would have different associated `taskInfo` schema, metadata, `web.monitor` functions, file names, etc. which should be introduced properly in the eventual refactor...